packages/forms/src/UIForm/fields/NestedListView/NestedListView.component.js

```diff
@@ -0,0 +1,259 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import keycode from 'keycode';
+import ListView from '@talend/react-components/lib/ListView';
+import { translate } from 'react-i18next';
+
+import { I18N_DOMAIN_FORMS } from '../../../constants';
+import getDefaultT from '../../../translate';
+import { getDisplayedItems, prepareItemsFromSchema } from './NestedListView.utils';
+import FieldTemplate from '../FieldTemplate';
+
+import theme from './NestedListView.scss';
+
+const DISPLAY_MODE_DEFAULT = 'DISPLAY_MODE_DEFAULT';
+const DISPLAY_MODE_SEARCH = 'DISPLAY_MODE_SEARCH';
+const DEFAULT_ITEM_HEIGHT = 33;
+
+const getItemHeight = () => DEFAULT_ITEM_HEIGHT;
+
+class NestedListViewWidget extends React.Component {
+	constructor(props) {
+		super(props);
+
+		const { t, value, schema, id } = props;
+
+		this.defaultHeaderActions = [
+			{
+				id: `${id}-search`,
+				icon: 'talend-search',
+				label: t('LISTVIEW_WIDGET_SEARCH', { defaultValue: 'Search for specific values' }),
+				onClick: this.switchToSearchMode.bind(this),
+			},
+		];
+
+		this.defaultSearchHeaderActions = [
+			{
+				id: 'abort',
+				icon: 'talend-cross',
+				label: t('LISTVIEW_WIDGET_ABORT', { defaultValue: 'Abort' }),
+				onClick: this.switchToDefaultMode.bind(this),
+			},
+		];
+
+		this.onInputKeyDown = this.onInputKeyDown.bind(this);
+		this.onInputChange = this.onInputChange.bind(this);
+
+		const searchCriteria = null;
+
+		const callbacks = {
+			onExpandToggle: this.onExpandToggle.bind(this),
+			onParentChange: this.onParentChange.bind(this),
+			onCheck: this.onCheck.bind(this),
+		};
+
+		this.items = prepareItemsFromSchema(schema, callbacks);
+		this.value = value;
+
+		this.state = {
+			searchCriteria,
+			displayedItems: getDisplayedItems(this.items, this.value, searchCriteria),
+		};
+	}
+
+	/**
+	 * Click callback to toggle the expanded status of children elements
+	 * @param {Object} event
+	 * @param {Object} item
+	 */
+	onExpandToggle(event, item) {
+		this.setState(({ searchCriteria }) => {
+			this.items = this.items.map(fieldItem => {
+				if (fieldItem.key === item.key) {
+					return { ...fieldItem, expanded: !fieldItem.expanded };
+				}
+				return fieldItem;
+			});
+
+			return {
```

Comment:
Only the `return` can be kept inside the `setState`, all the above can be extracted before it.